pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go

```diff
@@ -254,6 +254,20 @@ func (n *clusterManagerController) sync(ctx context.Context, controllerContext f
 		currentGenerations = append(currentGenerations, currentGeneration)
 	}
 
+	// Generate relatedResources from the resource files.
+	manifestsFiles := append(staticResourceFiles, deploymentFiles...)
```

Comment:
could we just add the related resource during apply? instead of reading the manifest again in apply

Reply:
done. add the related resource during the apply.